pricegraph/src/orderbook.rs

```diff
@@ -271,6 +270,56 @@ impl Orderbook {
         Some(invert_price(last_transient_price))
     }
 
+    /// Fill an order at a given price, returning the estimated buy amount that
+    /// can be filled at the given price with the current existing orders.
+    ///
+    /// This is calculated by repeatedly finding the cheapest path between a
+    /// token pair that is below the specified price and adding the capacity of
+    /// the path to the result.
+    ///
+    /// Note that the limit price is expressed as an exchange limit price, i.e.
+    /// with implicitely included fees.
+    pub fn fill_order_at_price(&mut self, pair: TokenPair, actual_limit_price: f64) -> f64 {
```

Comment:
Nice, easy to follow method! Is this what you are referring to as spaghetti code? Looks alright to me.

Reply:
This method reads fine, its more about having so many utility methods with implicit requirements.